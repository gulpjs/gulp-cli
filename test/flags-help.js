'use strict';

var lab = exports.lab = require('lab').script();
var code = require('code');

var fs = require('fs-extra');
var child = require('child_process');

var path = require('path');
var outfile = path.resolve(__dirname, 'output/flags-help.out');

var output = fs.readFileSync(__dirname + '/expected/flags-help.txt', 'utf8').replace(/(\r\n|\n|\r)\s?/gm,'\n');

lab.experiment('flag: --help', function() {

  lab.before(function(done) {
    fs.mkdirpSync(path.resolve(__dirname, 'output'));
    done();
  });

  lab.after(function(done) {
    fs.removeSync(path.resolve(__dirname, 'output'));
    done();
  });

  lab.test('shows help using --help', function(done) {
    child.exec('node ' + __dirname + '/../bin/gulp.js --help --cwd ./test/fixtures/gulpfiles > ' + outfile, function(err) {
      var stdout = fs.readFileSync(outfile, { encoding: 'utf8' });
      code.expect(stdout.replace(/(\r\n|\n|\r)\s?/gm,'\n')).to.equals(output);
      done(err);
    });
  });

  lab.test('shows help using short --h', function(done) {
    child.exec('node ' + __dirname + '/../bin/gulp.js --h --cwd ./test/fixtures/gulpfiles > ' + outfile, function(err) {
      var stdout = fs.readFileSync(outfile, { encoding: 'utf8' });
      code.expect(stdout.replace(/(\r\n|\n|\r)\s?/gm,'\n')).to.equals(output);
      done(err);
    });
  });

});
