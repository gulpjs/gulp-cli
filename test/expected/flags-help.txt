
Usage: gulp [options] tasks

Options:
  --help, -h       Show this help.  [boolean]
  --version, -v    Print the global and local gulp versions.  [boolean]
  --require        Will require a module before running the gulpfile. This is useful for transpilers but also has other applications.  [string]
  --gulpfile       Manually set path of gulpfile. Useful if you have multiple gulpfiles. This will set the CWD to the gulpfile directory as well.  [string]
  --cwd            Manually set the CWD. The search for the gulpfile, as well as the relativity of all requires will be from here.  [string]
  --verify         Will verify plugins referenced in project's package.json against the plugins blacklist.
  --tasks, -T      Print the task dependency tree for the loaded gulpfile.  [boolean]
  --depth          Specify the depth of the task dependency tree.
  --tasks-simple   Print a plaintext list of tasks for the loaded gulpfile.  [boolean]
  --tasks-json     Print the task dependency tree, in JSON format, for the loaded gulpfile.
  --color          Will force gulp and gulp plugins to display colors, even when no color support is detected.  [boolean]
  --no-color       Will force gulp and gulp plugins to not display colors, even when color support is detected.  [boolean]
  --silent, -S     Suppress all gulp logging.  [boolean]
  --continue       Continue execution of tasks upon failure.  [boolean]
  --log-level, -L  Set the loglevel. -L for least verbose and -LLLL for most verbose. -LLL is default.  [count]

